### ISSUE TITLE
Bug: Dialog does not properly take focus 

### ISSUE THREAD

**Issue by BigMacAdmin** [swiftDialog/swiftDialog#387]
**Describe the bug**
Inconsistently, but regularly, I will find that a Dialog window does not properly "activate" even when it has taken focus.

This happens frequently if Terminal is not the active window when the dialog draws. For my example I'm using Shortcuts to kick off a script, but I observe this when running from management tools (like Self Service) or LaunchAgents (Renew).

When in this bugged state, Dialog is the currently focused window, but the cursor will not appear in the active text field and shortcuts like cmd+v do not work. I can tab between text fields, but the cursor will not indicate which is active. Also, the button will appear as if the dialog is _not_ the active window.

Clicking another application and then back to the dialog fixes the problem, but when using `--blurscreen` this presents a challenge, especially if a dialog has multiple elements like checkboxes/radio buttons and text fields.

This bug has existed for months (maybe years?) and you spoiled me by fixing my other issues too fast, so now I'm giving you this one.

**To Reproduce**
1. Create a dialog window with one or more text fields which will trigger by some action other than Terminal
     a. Alternatively, use Terminal but add a `sleep 3` before your dialog command and be sure another application is focused
4. Activate Dialog and observe whether the bug has occurred. If not, close the dialog and try again. In my testing it happens at least 1 out of 4 times.

Here is the example script I used in the video below:
```bash
dialog \
--title "Focus?" \
--message "Every few times a script runs, I will find that: \n\n- Buttons will not have accent color\n\n- Cursor will not appear\n\n- Cannot paste using cmd-v\n\n- No amount of clicking will focus the Dialog window, until I choose a different window to focus first\n\n- That's tricky when using --blurscreen" \
--textfield "asdf" \
--textfield "asdf2" \
--height "600" \
--ontop \
--moveable \
--icon none

```

**Expected behavior**
Dialog properly taking focus and text fields indicating etc.

**Screenshots**

https://github.com/swiftDialog/swiftDialog/assets/106293503/d79cbf23-2bdb-484f-b0d2-d353f60fa3c7

**Desktop (please complete the following information):**
 - macOS version 14.5
 - swiftDialog version `2.4.2.4755` and `2.5.0-4763` but has existed for quite some time previously

**Comment by bartreardon** [comment]
I'm not able to replicate in any of the version I have. Are you running any other tools that affect window behaviour like a tiling manager or the like? Hard to know what's going on.

That said, I've moved the "activate" code to happen after the window placement code and we can see if that helps.

**Comment by garrettthorn** [comment]
Yes, I've been seeing the same behavior in my environment exactly as you describe.  I've only seen this when running scripts from Jamf Self Service.

It is quite infrequent, but it happens enough that I've been asked about it from a few of our users.

**Comment by cocopuff2u** [comment]
FYI confirmed i saw this bug with swiftDialog `2.5.5.4802` on MacOS `15.5`, only notice it if i open and close the scripts quickly during testing

**Comment by wakco** [comment]
Just to add to this, the activate code does not appear to work when the dialog is running over the login window (v2.5.6.4805 on both macOS 15.6.1 Sequoia and macOS 26 Tahoe Beta 6/7, that I have tested so far).

I have my own version of Baseline/Setup-Your-Mac and it display a dialog over the loginwindow (after Setup Assistant), then re-displays it after a restart and automatic login so it can display the same information, with the `activate:` command being sent to keep the dialog in the foreground, I've noticed the window dim and never re-activate before the restart (over the login window), only after the restart/automatic-login does the `activate:` command start working.

**Comment by bartreardon** [comment]
checking in to see if this is still an issue in 3.1.0 (coming up a year old)
